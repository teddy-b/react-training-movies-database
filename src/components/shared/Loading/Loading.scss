.loading {
  display: flex;
  justify-content: center;
  margin-top: 40vh;
}
