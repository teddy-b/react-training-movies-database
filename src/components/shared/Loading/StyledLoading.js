import styled from 'styled-components'

const StyledLoading = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 20vh;
`

export default StyledLoading
