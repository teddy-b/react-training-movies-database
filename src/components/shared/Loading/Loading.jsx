/* @flow */

import * as React from 'react'
import ReactLoading from 'react-loading'

import StyledLoading from './StyledLoading'

const Loading = (): React.Node => (
  <StyledLoading>
    <ReactLoading type="spinningBubbles" color="deepSkyBlue" />
  </StyledLoading>
)

export default Loading
