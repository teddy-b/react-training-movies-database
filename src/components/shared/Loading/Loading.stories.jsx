/* @flow */

import * as React from 'react'

import { storiesOf } from '@storybook/react'

import type { Renderable } from '@storybook/react'

import Loading from './Loading'

storiesOf('Components/Loading', module)
  .add('default', (): Renderable => (
    <Loading />
  ))
