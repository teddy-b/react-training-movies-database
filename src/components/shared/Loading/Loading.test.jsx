import React from 'react'
import renderer from 'react-test-renderer'

import Loading from './Loading'

jest.mock('react-loading', () => 'ReactLoading')

describe('Loading', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Loading />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
