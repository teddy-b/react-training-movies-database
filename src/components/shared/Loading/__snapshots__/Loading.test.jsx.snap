// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
<div
  className="sc-bdVaJa emJJEn"
>
  <ReactLoading
    color="deepSkyBlue"
    type="spinningBubbles"
  />
</div>
`;
